### PR TITLE
dup c_str before return

### PR DESCRIPTION
**Pull request by wangfenjin**
Otherwise memory might  corrupt

### CODE REVIEW

**Comment by Mytherin** [comment]
Thanks for the PR! Looks good, but could you fix the leak errors in the tests?

**Comment by Mytherin** [comment]
Thanks, looks good!

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -4,13 +4,10 @@ include_directories(../third_party/sqlite/include)
 include_directories(include)
 
 add_subdirectory(api)
-add_subdirectory(appender)
-add_subdirectory(common)
-add_subdirectory(extension)
+# add_subdirectory(appender) add_subdirectory(common)
```
Comment (hannesmuehleisen) [inline test/CMakeLists.txt]: Was this intentional?!

Comment (wangfenjin) [reply test/CMakeLists.txt]: No I already rollback this change. It's for speed up the local testing 😄

Comment (Mytherin) [reply test/CMakeLists.txt]: You can run individual tests using `build/debug/test/unittest "Basic test of C API"`, or run all the capi tests using `build/debug/test/unittest "[capi]"`. No need to disable the compilation of the other tests :)

Comment (wangfenjin) [reply test/CMakeLists.txt]: OK, thanks!

**Pull request merged**